scipy/spatial/transform/rotation.py

```diff
@@ -1460,3 +1460,127 @@ def __getitem__(self, indexer):
                [ 0.57735027,  0.57735027, -0.57735027,  0.        ]])
         """
         return self.__class__(self._quat[indexer], normalized=True)
+
+
+class Slerp(object):
+    """Spherical Linear Interpolation of Rotations.
+
+    The interpolation between consecutive rotations is performed as a rotation
+    around a fixed axis with a constant angular velocity [1]_. This ensures
+    that the interpolated rotations follow the shortest path between initial
+    and final orientations.
+
+    Parameters
+    ----------
+    times : array_like, shape (N,)
+        Times of the known rotations. At least 2 times must be specified.
+    rotations : `Rotation` instance
+        Rotations to perform the interpolation between. Must contain N
+        rotations.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Slerp
+            <https://en.wikipedia.org/wiki/Slerp#Quaternion_Slerp>`_
```

Comment:
Perhaps worth noting that I also have a Cython implementation of [geometric slerp in an open PR](https://github.com/scipy/scipy/pull/8069/files#diff-e8041eccf45474d9439678249b4f43b2R38). Not sure if any of the careful optimizations used there are applicable for the quaternion scenario.

Reply:
It is unfortunate this comment wasn't discussed, but I maintain that [geometric slerp](https://en.wikipedia.org/wiki/Slerp#Geometric_Slerp) is more general than quaternion slerp because it isn't constrained to 3 dimensions and the quaternion infrastructure. I'll likely open a PR proposing something like `scipy.spatial.geometric_slerp` with a `See Also` section that points to the quaternion / 3D-only version.